runhouse/resources/hardware/cluster.py

```diff
@@ -637,7 +637,7 @@ def _sync_default_env_to_cluster(self):
         env_vars = _process_env_vars(self._default_env.env_vars)
         setup_cmds = self._default_env.setup_cmds
 
-        for node in self.ips:
+        for node in self.ips or []:
```

Comment:
any issue with just doing this once in the constructor: `self.ips = ips or []`?

also i think with this change we'd also have to update the `head_node` property to avoid an IndexError

Reply:
I think we could change the constructor but then we'd have to enforce that self.ips is never set to `None` in other places (which it can be). Feels like we should do this as a safety measure regardless?

Reply:
and the `head_node` property is fine:
```
@property
def head_ip(self):
    return self.ips[0] if isinstance(self.ips, List) else None
```

Reply:
Actually nvm, that method was problematic too, so it's updated now!